tune_sklearn/tune_basesearch.py

```diff
@@ -253,10 +254,19 @@ def __init__(self,
                  return_train_score=False,
                  local_dir="~/ray_results",
                  max_iters=1,
-                 use_gpu=False):
+                 use_gpu=False,
+                 pipeline_detection=True):
         if max_iters < 1:
             raise ValueError("max_iters must be greater than or equal to 1.")
         self.estimator = estimator
+        self.base_estimator = estimator
+        self.base_estimator_name = ""
+        self.pipeline_detection = pipeline_detection
+
+        if self.pipeline_detection and isinstance(self.base_estimator,
+                                                  Pipeline):
+            self.base_estimator_name, self.base_estimator = self.base_estimator.steps[
```

Comment:
so pipeline_detection is only useful for the case where the user is using 1. early stopping AND 2. is using a pipeline right?

You would never want - "early stopping = True AND classifier=pipeline AND pipeline_detection=False" right? That'd be the same as:

"early stopping = False AND using pipeline"

If the above is true, then maybe we can just get rid of `pipeline_detection` and just automatically use _PipelineTrainable if "classifier=pipeline AND early stopping = True"

Let me know if that's not right!

Reply:
Basically, my intention was to allow the user to choose to opt out of this if they, for example, are using a custom Pipeline that eg. implements `partial_fit` or needs to be handled in some other way. That's why I have it set to True by default, but there's still the option not to do it.

Reply:
OK got it; I guess that's fair. I was just worried about having too many parameters in the initialization. 

What do you think about renaming `pipeline_detection` to `fit_last_stage=True` (and say this is only relevant if the estimator is a Pipeline object)?